CustomTextAnalytics.CognitiveSearch/Solution/CustomTextAzureFunction/CustomSkillsetFunction/Function.cs

```diff
@@ -43,49 +44,37 @@ public static class Function
             var customtextEndpoint = request.Headers["endpointUrl"];
             var customtextKey = request.Headers["apiKey"];
-            var customtextProjectName = request.Headers["projectName"];
-            var customtextDeploymentName = request.Headers["deploymentName"];
+            var recognizeCustomEntitiesProjectName = request.Headers["projectName"];
+            var recognizeCustomEntitiesDeploymentName = request.Headers["deploymentName"];
 
-            if (string.IsNullOrEmpty(customtextEndpoint) || string.IsNullOrEmpty(customtextEndpoint) || string.IsNullOrEmpty(customtextEndpoint) || string.IsNullOrEmpty(customtextEndpoint))
+            if (string.IsNullOrEmpty(customtextEndpoint) || string.IsNullOrEmpty(customtextKey) || string.IsNullOrEmpty(recognizeCustomEntitiesProjectName) || string.IsNullOrEmpty(recognizeCustomEntitiesDeploymentName))
             {
                 return new BadRequestObjectResult("One of or more of Custom Text app secrets wasn't provided in the request headers!");
             }
 
-            // init sdk client
-            var customTextClient = new CustomTextAnalyticsClient(customtextEndpoint, customtextKey);
-
-            // process input data
-            var response = new CustomSkillsetResponse();
-            var responseErrors = new StringBuilder();
-
-            foreach (var inRecord in requestRecords)
+            try
             {
-                var outRecord = new ResponseRecord() { RecordId = inRecord.RecordId };
-                try
+                // create input documents
+                var inputDocuments = requestRecords.Select(inRecord =>
                 {
-                    var text = inRecord.Data["text"] as string;
-                    var entities = await customTextClient.AnalyzeCustomEntitiesAsync(text, customtextProjectName, customtextDeploymentName);
-                    var groupedEntities = entities.GroupBy(entity => entity.Category);
+                    return new TextDocumentInput(inRecord.RecordId, inRecord.Data["text"] as string);
+                }).ToList();
 
-                    foreach (var group in groupedEntities)
-                    {
-                        outRecord.Data.Add(group.Key, group.Select(x => x.Text).ToList());
-                    }
-                }
-                catch (Exception e)
-                {
-                    outRecord.Errors.Add(new ErrorWarning() { Message = $"Error processing the request record : {e.Message}" });
-                }
+                // init custom text service
+                var analyzeCustomTextService = new AnalyzeCustomTextService(customtextEndpoint, customtextKey);
 
-                response.Values.Add(outRecord);
-            }
+                // analyze custom text
+                var responseRecords = await analyzeCustomTextService.AnalyzeCustomText(inputDocuments, recognizeCustomEntitiesProjectName, recognizeCustomEntitiesDeploymentName);
```

Comment:
Consider that there will be 2 more actions to pass their project name and deployment name here, I don't think it will be a great idea to pass all of them as a parameters. 

I am suggesting one of the two solutions: 
- Pass the whole request headers
- make them public in the service class

Reply:
@mshaban-msft